tasks/Common.cpp

```diff
@@ -314,9 +345,32 @@ GstFlowReturn Common::sinkNewSample(GstElement* sink, Common::ConfiguredOutput*
     else {
         std::memcpy(pixels, mapInfo.data, frame->getNumberOfBytes());
     }
-    data->frame.reset(frame.release());
-    data->port->write(data->frame);
-    return GST_FLOW_OK;
+    return frame;
+}
+
+std::unique_ptr<base::samples::frame::Frame> Common::sinkCompressedFrame(
+    GstMapInfo& mapInfo,
+    GstVideoInfo& videoInfo,
+    Common::ConfiguredOutput* data,
+    std::unique_ptr<Frame> frame)
+{
+    int width = videoInfo.width;
+    int height = videoInfo.height;
+
+    frame->init(width, height, 8, data->frameMode, 0UL, mapInfo.size);
+
+    if (frame->getNumberOfBytes() > mapInfo.size) {
+        data->task->queueError(
+            "Inconsistent number of bytes. Rock's image type calculated " +
+            to_string(frame->getNumberOfBytes()) + " while GStreamer only has " +
+            to_string(mapInfo.size));
+        return nullptr;
+    }
```

Comment:
you just explicitly set getNumberOfBytes to mapInfo.size. This check is not needed here.